paimon-core/src/main/java/org/apache/paimon/utils/BucketRecordAttributeManager.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.utils;
+
+/**
+ * Manager of the attributes of the internal records during runtime.
+ *
+ * <p>Different from {@link RecordAttributeManager}, this class manages the attributes need to be
```

Comment:
Why it need to be bucket's granularity?

Reply:
Because the mergeTreeWriter and its flushing behavior is performed at bucket's granularity. From the Flink operator's perspective, it is not sure whether a batch of data flushed in MergeTreeWriter is all insert-only or not.

Reply:
According to the latest comment, we agree to introduce `withInsertOnly` instead of this class. Thus this comment is not applicable now.